### DISCUSSION TITLE
Should we change something to limit the amount of breaking Changes on v19?

### DISCUSSION THREAD

**Discussion by Flyingmana** [OpenMage/magento-lts#2886] [Polls]
Over the last months we had several breaking changes merged into v19.
Some of them are bigger, some of them are smaller.

examples(I will extend this list continuously):
* https://github.com/OpenMage/magento-lts/pull/2242
* https://github.com/OpenMage/magento-lts/pull/2810
*

**Comment by tmotyl** [comment]
We should change the process to work mainly on v20, and only cherry-pick important bug fixes and security changes to v19.

**Reply by fballiano** [reply]
@tmotyl please vote for https://github.com/OpenMage/rfcs/pull/6

**Reply by tmotyl** [reply]
Not sure what do you mean

**Comment by fballiano** [comment]
why is https://github.com/OpenMage/magento-lts/pull/2810 considered BC?

**Reply by sreichel** [reply]
Think @Flyingmana means changed translateable string. This can be exlcuded an reverted in v20.

**Reply by fballiano** [reply]
we need an RFC to decide once and for all how to handle changes in the translatable strings, although we already broke all of the translation packs multiple times so...

but still, it's a bit of a stretch to consider that a BC, it technically is but then half of the merged PRs are BC :-D

**Comment by sreichel** [comment]
I'd ask another question ...

_Should we change something to limit the amount of breaking Changes on __v20__?_

I think we only have 3-4 real BC breaking PRs (IE8, ...). Others could be written w/o breaking anything. It was no fun to spent time for #2854.

#2810 should be only a small PR to remove some diffs between branches and now causes trouble. (comments/deprecation notes also belong to v19. Change to that translatable should not had be added at all. Minor changes (to v20 only) that make things more complicate.

#1257 seems to be more fix, rather a bc break

In rare cases we can use `Mage::getOpenMageMajorVersion() === 20` to _sperate_ code/functionality.

Last one ... IE support. I think we dont break the _"LTS backward compatibility promise"_ when we remove that. Its 2023.

**Comment by sreichel** [comment]
And ... we have some PRs that brings no benefits, but have potential to break something. E.g. 

- OpenMage/magento-lts#1551
- OpenMage/magento-lts#2859

We can mark every method/class as deprecated, but keep it for now.

It does not hurt and does not break anything.

**Reply by fballiano** [reply]
I think they have the benefit of cleaning the code for developers that are studying it to use it and they can get confused by the amount of dead files and code.
It should also make the tests (phpstan etc) run on less useless code sparing time and resources.

The right way would be deprecating it (but then it never gets removed) or removing it from v20 (but we have the whole other topic there).

But it's good because it's sparks the conversation ;-)

**Reply by sreichel** [reply]
You know i agree with removing dead code, but we're just talking about a few files here,

I dont know if the other autoloaders are used in extensions.  (Also dont know any extensions that did). Removing admin routing can break old extensions that still work.

> The right way would be deprecating it (but then it never gets removed) or removing it from v20

I'd not even remove them from v20 ... this should be done in next major release.

**Comment by sreichel** [comment]
v19 got no love for months and is EOL with april.
